submission-folder/liyuan.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * HOMEWORK
+ * rename this file to *your name*.ts
+ * Translate this JS code into valid TypeScript that follows good practices,
+ * following instructions in the comments
+ */
+
+// replace all `var` declarations appropriatly
+
+const NINE_THOUSAND : number = 9000
+let number : number = 6
+number = 7
+
+// create a type alias (rmb type alias names use PascalCase)
+
+type Student = {
+    name: string,
+    year: number,
+    friends: string[]
+    favouriteVariable: string|number,
+}
+
+let bob: Student = {
+  name: 'Bob',
+  year: 2,
+  friends: ['Steve'],
+  favouriteVariable: 23,
+}
+
+let steve: Student = {
+  name: 'Steve',
+  year: 1,
+  friends: ['Bob', 'Mary'],
+  favouriteVariable: 'hello', // note that this shall ONLY be string or number
+}
+
+let mary: Student = {
+  name: 'Mary',
+  year: 1,
+  friends: ['Steve'],
+  favouriteVariable: 100,
+}
+
+let students: Student[] = [bob, steve, mary] // assign an appropriate type to `students`
+
+// convert to anonymous function
+// use string literals
+// add appropriate type declarations to function args
+const printNameAndYear = (student: Student) => {
+  let text: String = `${student.name} is year  ${student.year}`
+  console.log(text)
+}
+const introduceSelf = (student: Student) => {
+  let text: String =
+    `Hello! I am ${student.name} and my favourite variable is ${student.favouriteVariable}`
+  console.log(text)
+}
+printNameAndYear(bob)
+introduceSelf(bob)
+printNameAndYear(steve)
+introduceSelf(steve)
+printNameAndYear(mary)
+introduceSelf(mary)
+
+// use short_circuit
+// rmb to replace all `var`
+
+let a: number = 1
+let b: string = 'bee'
+let c: number[] = [3, 4, 5]
+
+
+a === 1 && console.log('a is one')
+
+b.length > 0 && console.log('b is a non-empty string')
+
+c.length === 3 && c[0] === 3 && console.log('c is an array of length 3, and its first element is the number 3')
+
+
+// use ternary operator
+let s : String = 'is a loooooooooong string'
```

Comment:
same, no need to explicitly type all variables. Also, always use lowercase `string` for string types

Reply:
fixed in c1a1105cebd79ab481debf3da9c66a071f307d76